llvm/include/llvm/IR/Argument.h

```diff
@@ -71,8 +71,8 @@ class Argument final : public Value {
   FPClassTest getNoFPClass() const;
 
   /// If this argument has a range attribute, return the value range of the
-  /// argument. Otherwise, std::nullopt is returned.
-  std::optional<ConstantRange> getRange() const;
+  /// argument. Otherwise, nullptr is returned.
+  const ConstantRange *getRange() const;
```

Comment:
I'd prefer to leave this alone as well, to keep the APIs consistent.

Reply:
Yes agree was thinking about it when I hade the two ifs after each other but was to deep in save copies mode to stop and reflect. Pushed a new commit as there was so mush removed so a fix commit did not make sense.